### PR TITLE
typescriptifying

### PR DESCRIPTION
**Pull request by tristanhcole**
Converting additional reducers to typescript

* user 👤
* export

Converting typescript reducers to generics:
* message 💬
* organisation 🏢
* transfer usages

All reducers written in Typescript now use generics/action creators and follow the same pattern setup by @EDsCODE

### CODE REVIEW

File: app/client/sagas/userSagas.js [hunk 1]
```diff
@@ -154,17 +158,20 @@ function* resetPin({ payload }) {
...
-      message: reset_response.message
-    });
+    yield put(
+      MessageAction.addMessage({
+        error: false,
+        message: reset_response.message
+      })
+    );
   } catch (fetch_error) {
     const error = yield call(handleError, fetch_error);
```
Comment (tristanhcole) [inline app/client/sagas/userSagas.js:168]: we should consider adding the following code to the `handleError` function in utils.js: `store.dispatch(MessageAction.addMessage({ error: true, message: error.message }))`

**Comment by tristanhcole** [comment]
Includes fix for following error introduced in https://github.com/teamsempo/SempoBlockchain/commit/0b570d6f67f788ec39a0ec2cca92f4bbf433b603#comments PR #210

```
Error: call: argument fn is undefined or null
    at check (io-6de156f3.js?86aa:33)
    at validateFnDescriptor (io-6de156f3.js?86aa:455)
    at call (io-6de156f3.js?86aa:524)
    at showFlashMessage$ (messageSaga.js?6e97:13)
    at tryCatch (runtime.js?f3a2:65)
    at Generator.invoke [as _invoke] (runtime.js?f3a2:303)
    at Generator.prototype.<computed> [as next] (runtime.js?f3a2:117)
    at next (redux-saga-core.esm.js?42a9:1157)
    at currCb (redux-saga-core.esm.js?42a9:1251)
    at eval (redux-saga-core.esm.js?42a9:481)
eval @ index.js?69b4:3680

index.js?69b4:3680 The above error occurred in task onAddFlashMessage
    created by onAddFlashMessage
    created by takeEvery(ADD_FLASH_MESSAGE, onAddFlashMessage)
    created by rootSaga
    created by rootSaga
    created by rootSaga
Tasks cancelled due to error:
takeEvery(ADD_FLASH_MESSAGE, onAddFlashMessage)
```

**Comment by tristanhcole** [comment]
Marking as ready for review as starting to get quite large

File: app/client/sagas/userSagas.js (hunk 1 above)
Comment (enjeyw) [reply app/client/sagas/userSagas.js:159]: though that does presume we necessarily want to flash the error (sometimes it needs interpretation i think?). Guess we could also add that as an input arg...

File: app/client/components/user/SingleUserManagement.tsx [hunk 2]
```diff
@@ -66,15 +74,15 @@ class SingleUserManagement extends React.Component<Props> {
         custom_attributes: attr_dict,
         business_usage_name: businessUsage
       },
-      user_id
+      parseInt(user_id)
```
Comment (enjeyw) [inline app/client/components/user/SingleUserManagement.tsx:77]: This feels weird?
```  
const user_id = this.props.userId.toString();
...
parseInt(user_id)
```

F

File: app/client/sagas/exportSaga.js [hunk 3]
```diff
@@ -9,30 +9,27 @@ import {
...
-} from "../reducers/exportReducer.js";
+import { ExportAction } from "../reducers/export/actions";
+import { ExportActionTypes } from "../reducers/export/types";
 
 import { exportAPI } from "../api/exportAPI";
 
 function* newExport({ payload }) {
   try {
     const result = yield call(exportAPI, payload);
-    if (result.status === "success") {
```
Comment (enjeyw) [inline app/client/sagas/exportSaga.js:24]: a great PR would be to
CMD+F `== "`  and remove any string comparisons where possible

File: app/client/reducers/rootReducer.ts [hunk 4]
```diff
@@ -42,7 +42,7 @@ const appReducer = combineReducers({
   spreadsheetUpload,
   datasetSave,
   datasetList,
-  newExportRequest,
+  export: ExportReducer,
```
Comment (enjeyw) [inline app/client/reducers/rootReducer.ts:45]: wouldn't it make more sense just to follow the common pattern and name the reducer `export`?

**Comment by enjeyw** [comment]
> Marking as ready for review as starting to get quite large

Hahaha yeah there's like 5 seperate PRs in here :S

File: app/client/components/user/SingleUserManagement.tsx (hunk 2 above)
Comment (tristanhcole) [reply app/client/components/user/SingleUserManagement.tsx:77]: haha will fix

File: app/client/sagas/exportSaga.js (hunk 3 above)
Comment (tristanhcole) [reply app/client/sagas/exportSaga.js:24]: yeah, I actually think this is the last one in a reducer

File: app/client/reducers/rootReducer.ts (hunk 4 above)
Comment (tristanhcole) [reply app/client/reducers/rootReducer.ts:45]: i guess so, no strong opinion

**Comment by tristanhcole** [comment]
> I'm getting a bug that I can't repeat on dev... seems to be a generalised version of this:
> #237
> If i try inviting a new user, it adds fine, but then nothing else loads unless i hard-refresh...

Hm ok, at first glance it sounds like it is a generalized problem to do with #237 which was probably introduced in https://github.com/teamsempo/SempoBlockchain/pull/186. I'll take a look and see if I can implement a fix

**Comment by tristanhcole** [comment]
> > I'm getting a bug that I can't repeat on dev... seems to be a generalised version of this:
> > #237
> > If i try inviting a new user, it adds fine, but then nothing else loads unless i hard-refresh...
> 
> Hm ok, at first glance it sounds like it is a generalized problem to do with #237 which was probably introduced in #186. I'll take a look and see if I can implement a fix

Hot fix for this error https://github.com/teamsempo/SempoBlockchain/pull/242 @enjeyw

File: app/client/reducers/rootReducer.ts (hunk 4 above)
Comment (tristanhcole) [reply app/client/reducers/rootReducer.ts:45]: I think the reason I did it was `export` is not allowed and we've done that flow before for previous typescript reducers;

```
  transferUsages: TransferUsageReducer,
  organisations: OrganisationReducer,

```

Comment (enjeyw) [reply app/client/reducers/rootReducer.ts:45]: oh export not being allowed is a pretty good justification lol!

**Pull request merged**